packages/jaeger-ui/src/components/SearchTracePage/SearchResults/index.js

```diff
@@ -83,9 +84,16 @@ export const sortFormSelector = formValueSelector('traceResultsSort');
 
 export default class SearchResults extends React.PureComponent<SearchResultsProps> {
   props: SearchResultsProps;
-
   static defaultProps = { skipMessage: false };
 
+  constructor(props) {
+    super(props);
+    const { queryOfResults, diffCohort } = this.props;
+    const searchUrl = getUrl(stripEmbeddedState(queryOfResults));
+    const cohortIds = new Set(diffCohort.map(datum => datum.id));
+    this.state = { searchUrl, cohortIds };
```

Comment:
Currently, this state is not being updated when new props come in, so selecting result items for comparison is not working.

Seems like using [memoize-one](https://github.com/alexreardon/memoize-one), instead, would be less work than maintaining the state.